### PR TITLE
Refactor dev

### PR DESCRIPTION
**Pull request by brucepang**
1. Made naming convention consistent. Camel case for variable, lower case for file names.
2. Classes, like SourceConstructor and EventObject (not in use rn), are put into a seperate folder
3. Created an utils.js file to store service independent functions, like flatten array or calculate time difference.
4. Remove legacy codes/files, like some routes in route.js and service/eventbrite.js.
5. Add sorting to the pipeline. Right now only do sort by time. Easy to extend to other features. 
6. Add try catch block on reading data from API in case there are any missing field.
-bo

### CODE REVIEW

File: api/controller.js [hunk 1]
```diff
@@ -2,10 +2,7 @@
 
 var properties = require('../package.json')
 var distance = require('../service/distance');
```
Comment (leaspark) [inline api/controller.js:4]: Do we still need the distance service? What is its purpose?

File: api/controller.js [hunk 2]
```diff
@@ -15,20 +12,14 @@ var controllers = {
        }
        res.json(aboutInfo);
    },
+   
    get_distance: function(req, res) {
```
Comment (leaspark) [inline api/controller.js]: Same with this - is distance used for anything?

**Review by leaspark — APPROVED** [review]

File: api/controller.js (hunk 1 above)
Comment (brucepang) [reply api/controller.js:4]: Probably not, David put it there for test purpose i believe.

File: api/controller.js (hunk 2 above)
Comment (brucepang) [reply api/controller.js]: Good catch.  It probably would help to calculate the distance between user and event, after we figure out how to get events' location. And the distance can help to do the sorting.

**Comment by mavysavydav** [comment]
LGTM! Also feel free to take out the zipcode distance stuff that I originally used for testing

**Review by mavysavydav — APPROVED** [review]

**Pull request merged**